src/script/common/c_internal.cpp

```diff
@@ -29,6 +29,19 @@ std::string script_get_backtrace(lua_State *L)
 	return luaL_checkstring(L, -1);
 }
 
+std::string script_get_caller_file_line(lua_State *L, unsigned int offset)
+{
+	lua_Debug ar;
+
+	if (!lua_getstack(L, 2 + offset, &ar))
```

Comment:
there might be a case where you want the actual calling function, i propose `1 + offset`

Reply:
Is there a reason your code uses 2 rather than 1 here? It prints out the wrong position as detailed above

Reply:
since `log_deprecated` is called by a wrapper function(?) you'll want to get the function that called the wrapper, not the wrapper itself....
okay this turns out to be trickier as `log_deprecated` is also called directly from C++ in some cases

Reply:
Ah, that explains it. I'll have to pass an offset into log_deprecated and default to 0, unfortunately